pkg/odo/cli/url/delete.go

```diff
@@ -26,26 +26,31 @@ var (
 
 // URLDeleteOptions encapsulates the options for the odo url delete command
 type URLDeleteOptions struct {
-	*clicomponent.CommonPushOptions
+	*clicomponent.PushOptions
 	urlName            string
 	urlForceDeleteFlag bool
 	now                bool
 }
 
 // NewURLDeleteOptions creates a new URLDeleteOptions instance
 func NewURLDeleteOptions() *URLDeleteOptions {
-	return &URLDeleteOptions{CommonPushOptions: clicomponent.NewCommonPushOptions()}
+	return &URLDeleteOptions{PushOptions: clicomponent.NewPushOptions()}
 }
 
 // Complete completes URLDeleteOptions after they've been Deleted
 func (o *URLDeleteOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+
 	if experimental.IsExperimentalModeEnabled() {
+
 		o.Context = genericclioptions.NewDevfileContext(cmd)
 		o.urlName = args[0]
 		err = o.InitEnvInfoFromContext()
 		if err != nil {
 			return err
 		}
+		if o.now {
+			o.CompleteDevfilePath()
+		}
```

Comment:
Maybe perform the complete devfilePath action irrespective of the now flag?

Reply:
Updated